OpenRA.Game/Map/TileSet.cs

```diff
@@ -246,6 +246,9 @@ public byte GetTerrainIndex(string type)
 
 		public byte GetTerrainIndex(TerrainTile r)
 		{
+			if (!Templates.ContainsKey(r.Type))
```

Comment:
You can save a dictionary lookup by using TryGetValue.  See e.g. 965447a42c92aecdbeacef69be4f72a1478ef08b.